connect/api/src/main/java/org/apache/kafka/connect/data/Values.java

```diff
@@ -723,14 +723,22 @@ public static DateFormat dateFormatFor(java.util.Date value) {
         return new SimpleDateFormat(ISO_8601_TIMESTAMP_FORMAT_PATTERN);
     }
 
+    protected static boolean canParseSingleTokenLiteral(Parser parser, boolean embedded, String tokenLiteral) {
+        int startPosition = parser.mark();
+        if (parser.canConsume(tokenLiteral)) {
+            if (embedded || !parser.hasNext()) {
```

Comment:
Yeah, this section was reworked so a top-level string like `"true}"` would parse correctly as the string `"true}"` as opposed to the boolean literal `true`. The special case for `embedded` was necessary since the substring `"true}"` in the string `"{3: true}"` should be parsed as the boolean `true` and the map end delimiter, as opposed to just the string `"true}"`.

You've convinced me to not only add a comment here detailing the intended functionality, but also a few more tests making sure that whitespace doesn't mess with the parsing of top-level booleans (it doesn't, but still good to add a test now to prevent regression).